### ISSUE TITLE
Create an Alert for specific pod phase and aggregate multiple alerts of same type

### ISSUE THREAD

**Issue by saiyam1814** [rancher/rancher#24357]
Question/enhancement

Right now when we create an Alert it sends a notification for all the pod events. But Can we have it for a specific pod phase and also can we aggregate similar kinds of alerts and send out only one.

**Example1**- Suppose a Pod is in Running State and suddenly goes to CrahloopBackOff then a notification should be triggered but only once not that next time which pod state is checked it should send. 

**Example2** - Suppose a Pod is in Running but it goes out of memory then Alert should be triggered to the user saying it's out of memory and that also only once.

**Example3** - Suppose a Pod is running but Node goes down and that pod is killed and placed on some other Node, Alert should be triggered with a termination message that pod is getting killed and getting created on other  node  only once with 

**Basically send Alert on Pod Phase change Alerts with the specific event/termination message.**
